instrumentation/opentelemetry-instrumentation-flask/src/opentelemetry/instrumentation/flask/__init__.py

```diff
@@ -245,12 +245,25 @@ def response_hook(span: Span, status: str, response_headers: List):
 from typing import Collection
 
 import flask
+import opentelemetry.instrumentation.wsgi as otel_wsgi
 from packaging import version as package_version
 
-import opentelemetry.instrumentation.wsgi as otel_wsgi
 from opentelemetry import context, trace
 from opentelemetry.instrumentation.flask.package import _instruments
 from opentelemetry.instrumentation.flask.version import __version__
+
+try:
+    flask_version = flask.__version__
+except AttributeError:
+    try:
+        from importlib import metadata
+    except ImportError:
+        import importlib_metadata as metadata
+    try:
+        flask_version = metadata.version("flask")
+    except (ImportError, metadata.PackageNotFoundError):
+        flask_version = "unknown"
```

Comment:
I wonder if these would work for all versions of flask. Maybe we can get rid of the `flask.__version__` branch altogether?

Reply:
It works for all flask versions we support (>=1.0.0).
Only the most recent version, 3.0.0, will use importlib to get the version information.
Once we officially support only Python 3.8 and above, this code could be simpler.